src/components/Api.js

```diff
@@ -0,0 +1,162 @@
+export default class Api {
+  constructor(apiSettings) {
+    this._apiSettings = apiSettings;
+    this._headers = apiSettings.headers;
+    this._baseUrl = apiSettings.baseUrl;
+    this._userInfoUrl = `${this._baseUrl}/users/me`;
+    this._cardsUrl = `${this._baseUrl}/cards`;
+    this._avatarUrl = `${this._userInfoUrl}/avatar`;
+  }
+
+  _checkResponse(response) {
+    if (!response.ok) {
+      throw response;
+    }
+  }
+
+  async getInitialCards() {
+    try {
+      const res = await fetch(this._cardsUrl, {
+        headers: this._headers,
+      });
+      this._checkResponse(res);
+      const cardsArray = await res.json();
+      console.log(cardsArray);
+      return cardsArray;
+    } catch (err) {
+      console.log("Failed to get cards: ", err);
+      throw new Error(err.status);
+    }
+  }
+
+  //cardsArray is an array of cardata objects
+  async addArrayOfCards(cardsArray) {
+    cardsArray.forEach(async (cardData) => {
+      this.addCard(cardData);
```

Comment:
`addArrayOfCards` is not needed in `Api` because there should be only `api`-requests